Sources/MapboxMaps/Foundation/Extensions/Core/MapEvents.swift

```diff
@@ -68,274 +68,98 @@ import Foundation
-         * │       └── message - String
-         * └── cancelled - Boolean
-         * ```
-         */
-        case resourceRequest
-
-        // swiftlint:disable:next cyclomatic_complexity
-        public init?(rawValue: String) {
-            switch rawValue {
-            case MapEvents.mapLoaded:
-                self = .mapLoaded
-            case MapEvents.mapLoadingError:
-                self = .mapLoadingError
-            case MapEvents.mapIdle:
-                self = .mapIdle
-            case MapEvents.styleDataLoaded:
-                self = .styleDataLoaded
-            case MapEvents.styleLoaded:
-                self = .styleLoaded
-            case MapEvents.styleImageMissing:
-                self = .styleImageMissing
-            case MapEvents.styleImageRemoveUnused:
-                self = .styleImageRemoveUnused
-            case MapEvents.sourceDataLoaded:
-                self = .sourceDataLoaded
-            case MapEvents.sourceAdded:
-                self = .sourceAdded
-            case MapEvents.sourceRemoved:
-                self = .sourceRemoved
-            case MapEvents.renderFrameStarted:
-                self = .renderFrameStarted
-            case MapEvents.renderFrameFinished:
-                self = .renderFrameFinished
-            case MapEvents.cameraChanged:
-                self = .cameraChanged
-            case MapEvents.resourceRequest:
-                self = .resourceRequest
-            default:
-                return nil
-            }
-        }
-
-        public var rawValue: String {
-            switch self {
-            case .mapLoaded:
-                return MapEvents.mapLoaded
-            case .mapLoadingError:
-                return MapEvents.mapLoadingError
-            case .mapIdle:
-                return MapEvents.mapIdle
-            case .styleDataLoaded:
-                return MapEvents.styleDataLoaded
-            case .styleLoaded:
-                return MapEvents.styleLoaded
-            case .styleImageMissing:
-                return MapEvents.styleImageMissing
-            case .styleImageRemoveUnused:
-                return MapEvents.styleImageRemoveUnused
-            case .sourceDataLoaded:
-                return MapEvents.sourceDataLoaded
-            case .sourceAdded:
-                return MapEvents.sourceAdded
-            case .sourceRemoved:
-                return MapEvents.sourceRemoved
-            case .renderFrameStarted:
-                return MapEvents.renderFrameStarted
-            case .renderFrameFinished:
-                return MapEvents.renderFrameFinished
-            case .cameraChanged:
-                return MapEvents.cameraChanged
-            case .resourceRequest:
-                return MapEvents.resourceRequest
-            }
+    /// Typed events emitted by the SDK
+    struct Event<Payload: Decodable> {
```

Comment:
TBD:
I'm not quite sure that we must have a `Decodable` requirement in that place. It's not a part of the requirement for the Event, it's more implementation detail.

Reply:
As we discussed internally, there is no way to avoid this requirement. The only alternative is to write down a static list of each event and manual conversion plan which has no benefits over the `Decodable` requirement.

Reply:
Apparently, the compiler doesn't complain if this requirement is lifed. Payload is still restricted to be Decodable in `onNext/Every` methods so that is enough.

Reply:
Sounds like we should remove `: Decodable` requirement in `onNext/onEvery` and left a single on Event itself

Reply:
Good point, this seems to work as well :)